tests/test_env.py

```diff
@@ -0,0 +1,30 @@
+# SPDX-License-Identifier: MIT
+
+import os
+import subprocess
+import sys
+import sysconfig
+
+import build.env
+
+
+def test_isolated_environment(mocker):
+    with build.env.IsolatedEnvironment() as env:
```

Comment:
I think this test should run in an isolated python environment (one that has no pip/setuptools/etc installed) so there's really no chance things are being pulled from some other path on the python path. E.g. is this past failing if you change from ``sys. get_scheme_names`` to that manual thing you defined before? It should if it's an accurate test.

Reply:
That "manual thing" I had before was hardcoding the scheme names, it should work the same. Why would this be failing?

> really no chance things are being pulled from some other path on the python path

What things being pulled? There is nothing being pulled. In this test I check if the environment is correct, after that I check if pip is called correctly, but we never actually call pip. I look at the path and make sure it is correct.

For eg. I import `sysconfig` before entering the isolated environment and make sure `purelib` or `platlib` are not there.

```python
for path in ('purelib', 'platlib'):
    assert sysconfig.get_path(path) not in os.environ['PYTHONPATH'].split(os.pathsep)
```

Reply:
> I had before was hardcoding the scheme names, it should work the same.

It should have been failing on pypy, as pypy uses different scheme names then what you hardcoded. The fact that it did not, shows something is not totally correct in this sense. 

> For eg. I import `sysconfig` before entering the isolated environment and make sure `purelib` or `platlib` are not there.

To be fair I'm not sure this is enough. I'd expect we'd isolate from any potential PYTHONPATH the user might have set, or additional paths the user injects via user/sitecustomize.

Reply:
It shouldn't.

```
>>>> import sysconfig
>>>> sysconfig.get_path('purelib')
'/opt/pypy/site-packages'
>>>> sysconfig.get_path('platlib')
'/opt/pypy/site-packages'
```

If you read the code you'll notice I remove both the default paths and the different schemes.

https://github.com/FFY00/python-build/blob/e5efbcca2df30b7de5e14477cdbd4fb83f1d932c/build/env.py#L55

The test is correct and working as expected.